CMakeLists.txt

```diff
@@ -292,6 +292,10 @@ if (NOT USE_SIMD STREQUAL "")
             else ()
                 add_definitions ("-m${feature}")
             endif ()
+            if (feature STREQUAL "fma" AND CMAKE_COMPILER_IS_CLANG)
+                # Clang needs an explicit flag to make fma ops
+                add_definitions ("-ffp-contract=fast")
```

Comment:
I can kinda get you there, but imperfectly.

The place in the CMakeLists.txt where we parse the USE_SIMD options, in addition to setting -mfma, I could also -DFMA_ENABLED=1 -- that is, make my own preprocessor symbol for the one that is lacking.

Then, we could create a madd_aggressive() that would use fma instructions any time FMA_ENABLED is true, even if -ffp-contract=fast is not turned on, whereas if you further turned on -ffp-contract=fast you would get it potentially everywhere because the compiler would look for places to do it automatically.

Now, there are two problems with this approach:

1. While this would work for our own codebase, non-OIIO apps that liked to use fmath.h or simd.h for convenience would not end up with fma from ordinary madd_aggressive unless they also took the effort to set the special symbol and the correct compiler flags.

2. gcc currently has -ffp-contract=fast turned on by default any time -mfma is enabled, so you'd need to explicitly use -ffp-contract=off if you wanted to ensure that fma's are ONLY used for madd_aggressive and not anywhere else.

You asked earlier about -ffp-contract=on versus fast. The answer is that "on" is not currently implemented, and is thus the same as "off": https://gcc.gnu.org/onlinedocs/gcc/Optimize-Options.html